cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -1166,9 +1166,16 @@ arrow::Status VeloxShuffleWriter::splitFixedWidthValueBuffer(const velox::RowVec
 
     VS_PRINTLF(sizePerRow);
 
-    uint64_t preAllocRowCnt = options_.offheap_per_task > 0 && sizePerRow > 0
-        ? options_.offheap_per_task / sizePerRow / numPartitions_ >> 2
-        : options_.buffer_size;
+    //remove the cache memory size since it can be spilled.
+    //we can't use pool_->bytes_allocated() here because it's arrow::default_pool(). we should define a 
+    // pool for spill buffer, just like the pool for split buffer
+    // memLimit+=pool_->bytes_allocated();
+    memLimit += totalCachedPayloadSize();
+    //make sure split buffer uses 128M memory at least, let's hardcode it here for now
+    if (memLimit<128*1024*1024) memLimit=128*1024*1024;
```

Comment:
Let's make 128M an configurable value

Reply:
Is this required from this PR? Or I'd move to the topic assigning initial reservation to memory pools.